src/v/cluster/tx_registry_stm.h

```diff
@@ -31,6 +33,48 @@ namespace cluster {
 
 class tx_registry_stm final : public persisted_stm {
 public:
+    enum class batch_subtype : int32_t { tx_mapping = 1 };
+
+    struct hosted_txs
+      : serde::
+          envelope<hosted_txs, serde::version<0>, serde::compat_version<0>> {
+        tx_hash_ranges_set hash_ranges{};
+        absl::btree_set<kafka::transactional_id> excluded_transactions{};
+        absl::btree_set<kafka::transactional_id> included_transactions{};
```

Comment:
Do we relay on ordering here ? If not this may be replaced with `node_hash_set`

Reply:
ok, but why it's better?

Reply:
it has smaller memory footprint and faster lookup